src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java

```diff
@@ -156,6 +157,27 @@ void setGameMode_GameModeChanged_GameModeSet()
 	{
 		player.setGameMode(GameMode.CREATIVE);
 		assertEquals(GameMode.CREATIVE, player.getGameMode());
+		server.getPluginManager().assertEventFired(PlayerGameModeChangeEvent.class);
```

Comment:
We can be a bit more narrow here even and assert that the event was a change to creative mode using `PlayerGameModeChangeEvent#getGameMode()` or something within the `.assertEventFired(...)` call

Reply:
You mean something like that? 
`server.getPluginManager().assertEventFired(PlayerGameModeChangeEvent.class, gameModeChangeEvent -> gameModeChangeEvent.getNewGameMode() == GameMode.CREATIVE);
`

Reply:
Exactly, though `gameModeChangeEvent` seems a bit long O.o
Feel free to just shorten it to `e` or `event`

Reply:
Ayyy I mean I could break the line but ok I'll rename it. I don't like the short names xD